File: PaintedSkin/src/main/java/org/alee/component/skin/pack/ResourcesType.java
package org.alee.component.skin.pack;

import androidx.annotation.StringDef;

import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;

/**********************************************************
 *
 * @author: MY.Liu
 * @date: 2021/2/10
 * @description: 资源类型
 *
 *********************************************************/
public final class ResourcesType {
    /**
     * drawable资源类型
     */
    public static final String DRAWABLE = "drawable";
    /**
     * Mipmap 资源类型
     */
    public static final String MIPMAP = "mipmap";

    /**
     * color 资源类型
     */
    public static final String COLOR = "color";
    /**
     * ColorStateList
     */
    public static final String COLOR_STATE_LIST = "colorStateList";

    /**
     * string 资源类型
     */
    public static final String STRING = "string";

    /**
     * dimen 资源类型
     */
    public static final String DIMEN = "dimen";

    @StringDef({DRAWABLE,MIPMAP, COLOR, COLOR_STATE_LIST, STRING, DIMEN})
    @Retention(RetentionPolicy.SOURCE)
    public @interface Constraint {

    }

}

File: Library/PaintedSkin-Kotlin/src/main/java/org/alee/component/skin/constant/ResourcesType.kt
package org.alee.component.skin.constant

import androidx.annotation.StringDef

/**
 * Android 资源类型
 *
 * <p> 详细描述
 *
 * @author MingYu.Liu
 * created in 2023/8/30
 *
 */
@StringDef(
    ResourcesType.DRAWABLE,
    ResourcesType.MIPMAP,
    ResourcesType.COLOR,
    ResourcesType.COLOR_STATE_LIST,
    ResourcesType.STRING,
    ResourcesType.DIMEN,
)
@Retention(AnnotationRetention.SOURCE)
annotation class ResourcesType {
    companion object {
        /**
         * drawable资源类型
         */
        const val DRAWABLE = "drawable"

        /**
         * Mipmap 资源类型
         */
        const val MIPMAP = "mipmap"

        /**
         * color 资源类型
         */
        const val COLOR = "color"

        /**
         * ColorStateList
         */
        const val COLOR_STATE_LIST = "colorStateList"

        /**
         * string 资源类型
         */
        const val STRING = "string"

        /**
         * dimen 资源类型
         */
        const val DIMEN = "dimen"
    }
}
